File: app/src/main/java/io/agora/chatdemo/fcm/FCMMSGService.java
package io.agora.chatdemo.fcm;

import android.util.Log;

import androidx.annotation.NonNull;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;

import io.agora.chat.ChatClient;
import io.agora.chatdemo.DemoHelper;
import io.agora.util.EMLog;

public class FCMMSGService extends FirebaseMessagingService {
    private static final String TAG = "EMFCMMSGService";

    @Override
    public void onMessageReceived(RemoteMessage remoteMessage) {
        super.onMessageReceived(remoteMessage);
        if (remoteMessage.getData().size() > 0) {
            String message = remoteMessage.getData().get("alert");
            EMLog.i(TAG, "onMessageReceived: " + message);
            DemoHelper.getInstance().getNotifier().notify(message);
        }
    }

    @Override
    public void onNewToken(@NonNull String token) {
        super.onNewToken(token);
        EMLog.i(TAG, "onNewToken: " + token);
        ChatClient.getInstance().sendFCMTokenToServer(token);
    }
}

File: app/src/main/kotlin/io/agora/chatdemo/fcm/FCMMSGService.kt
package io.agora.chatdemo.fcm

import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import io.agora.chat.ChatClient
import io.agora.chatdemo.DemoHelper
import io.agora.util.EMLog

class FCMMSGService: FirebaseMessagingService() {
    private val TAG = "FCMMSGService"

    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        super.onMessageReceived(remoteMessage)
        if (remoteMessage.data.isNotEmpty()) {
            val message = remoteMessage.data["alert"]
            EMLog.i(TAG, "onMessageReceived: $message")
            DemoHelper.getInstance().getNotifier()?.notify(message)
        }
    }

    override fun onNewToken(token: String) {
        super.onNewToken(token)
        EMLog.i(TAG, "onNewToken: $token")
        ChatClient.getInstance().sendFCMTokenToServer(token)
    }
}